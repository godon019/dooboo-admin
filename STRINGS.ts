const STRINGS = {
  en: {
    // English
    BUTTON: 'Button',
    SIGNUP: 'Sign Up',
    ADMIN_LOGIN: 'Admin Login',
    LOGIN: 'Login',
    LOGOUT: 'Logout',
    GOOGLE_LOGIN: 'Login with Google',
    FACEBOOK_LOGIN: 'Login with Facebook',
    EMAIL: 'Email',
    EMAIL_HINT: 'Write email address.',
    EMAIL_ERROR: 'Wrong email address',
    PASSWORD: 'Password',
    PASSWORD_CONFIRM: 'Confirm Password',
    PASSWORD_FIND: 'Find Password',
    PASSWORD_HINT: 'Write password.',
    PASSWORD_NOT_MATCH: 'Password and confirm password does not match.',
    COMPLETE: 'Done',
    GOTO_Temp: 'No Page',
    BACK: 'Back',
    NAVIGATE: 'Navigate',
    CHANGE_THEME: 'Change theme',
    SIGN_IN_SCREEN: 'Sign in screen',
    SIGN_UP_SCREEN: 'Sign up screen',
    MAIN_SCREEN: 'Main screen',
    UPDATE_BUTTON: 'Update',
    DELETE_BUTTON: 'Delete',
    SERVICE_LIST: 'Service List',
    CATEGORY_LIST: 'Category List',
    SUBS_LIST: 'Subs List',
    SUBS_ADD_BUTTON: '+ Add Subs',
    SAVE: 'Save',
    SUBMIT: 'Submit',
    CANCEL: 'Cancel',
    // Service Screen
    SERVICE_ADD_BUTTON: '+ Add Service',
    // Category Screen
    CATEGORY_ADD_BUTTON: '+ Add Category',
    // Service Modal
    ICON: 'Icon',
    THUMBNAIL: 'Thumbnail',
    IMAGE: 'Image',
    SERVICE_NAME_KEY: 'name key',
    WEB_PAGE: 'Web Url',
    ANDROID_PLAY_STORE_URL: 'Android url',
    IOS_APP_STORE_URL: 'IOS url',
    MEMO: 'Memo',
    PNG_JPG_GIF_ONLY: 'you can upload png, jpg and gif files only',
    INPUT_REQUIRED: 'is required',
    INPUT_MAX_VALIDATION: 'you exceeded max length of',
    INPUT_TEXT_TO_SEARCH: 'Search.',
    // Service List
    SERVICE_LIST_TABLE_LABEL_ID: 'ID',
    SERVICE_LIST_TABLE_LABEL_NAME: 'SERVICE_EN',
    SERVICE_LIST_TABLE_LABEL_NAME_KR: 'SERVICE_KR',
    SERVICE_LIST_TABLE_LABEL_THUMBNAIL: 'THUMBNAIL',
    SERVICE_LIST_TABLE_LABEL_SETTING: 'SETTING',
    PREVIEW_ICON: 'Preview',
    // Category List
    CATEGORY_LIST_TABLE_LABEL_ID: 'ID',
    CATEGORY_LIST_TABLE_LABEL_KEY: 'KEY',
    CATEGORY_LIST_TABLE_LABEL_DESCRIPTION: 'DESCRIPTION',
    CATEGORY_LIST_TABLE_LABEL_RELEASED: 'Released',
    CATEGORY_LIST_TABLE_LABEL_SETTING: 'SETTING',
    // Service Group
    SERVICE_GROUP_ADD_BUTTON: '+ Add Service Group',
    SERVICE_GROUP_LIST: 'Service Group List',
    SERVICE_GROUP_LIST_TABLE_LABEL_ID: 'ID',
    SERVICE_GROUP_LIST_TABLE_LABEL_KEY: 'KEY',
    SERVICE_GROUP_LIST_TABLE_LABEL_DESCRIPTION: 'DESCRIPTION',
    SERVICE_GROUP_LIST_TABLE_LABEL_SETTING: 'SETTING',
    // Service Details
    SERVICE_LABEL_ID: 'ID',
    SERVICE_LABEL_NAME: 'KEY NAME',
    SERVICE_LABEL_PRICE_PLAN_CURRENCY: 'CURRENCY',
    SERVICE_LABEL_PRICE: 'PRICE',
    SERVICE_LABEL_PERIOD_UNIT: 'PERIOD UNIT',
    SERVICE_LABEL_PERIOD_TYPE: 'PERIOD TYPE',
    SERVICE_LABEL_SETTING: 'SETTING',
    // Subs modal
    SUBS_MODAL_TITLE: 'ADD SUBS',
    SUBS_MODAL_NAME: 'NAME',
    SUBS_MODAL_CURRENCY: 'CURRENCY',
    SUBS_MODAL_PRICE: 'PRICE',
    SUBS_MODAL_PAYMENT_DATE_UNIT: 'PAYMENT DATE UNIT',
    SUBS_MODAL_CASE: 'CASE',
    SUBS_MODAL_MEMO: 'MEMO',
    // Category modal
    CATEGORY_MODAL_CREATE_TITLE: 'Add Category',
    CATEGORY_MODAL_UPDATE_TITLE: 'Update Category',
    CAT_MODAL_KEY: 'key',
    CAT_MODAL_DESC: 'description',
    CAT_MODAL_RELEASED: 'Released',
    // Service Group modal
    SERVICE_GROUP_MODAL_CREATE_TITLE: 'Add Service Group',
    SERVICE_GROUP_MODAL_UPDATE_TITLE: 'Update Service Group',
    SERVICE_GROUP_MODAL_KEY: 'key',
    SERVICE_GROUP_MODAL_DESC: 'description',
  },
  ko: {
    // Korean
    BUTTON: '버튼',
    SIGNUP: '회원가입',
    ADMIN_LOGIN: '어드민 로그인',
    LOGIN: '로그인',
    LOGOUT: '로그아웃',
    GOOGLE_LOGIN: '구글 로그인',
    FACEBOOK_LOGIN: '페이스북 로그인',
    EMAIL: '이메일',
    EMAIL_ERROR: '이메일이 잘못 입력되었습니다.',
    EMAIL_HINT: '이메일 주소를 입력해주세요.',
    PASSWORD: '비밀번호',
    PASSWORD_CONFIRM: '비밀번호 확인',
    PASSWORD_FIND: '비밀번호 찾기',
    PASSWORD_HINT: '비밀번호를 입력해주세요.',
    PASSWORD_NOT_MATCH: '입력된 비밀번호와 비밀번호 확인이 일치하지 않습니다.',
    COMPLETE: '완료',
    GOTO_Temp: '없는 페이지',
    BACK: '뒤로',
    NAVIGATE: '이동하기',
    CHANGE_THEME: '테마 변경',
    SIGN_IN_SCREEN: '로그인 화면',
    SIGN_UP_SCREEN: '회원가입 화면',
    MAIN_SCREEN: '메인 화면',
    UPDATE_BUTTON: '수정',
    DELETE_BUTTON: '삭제',
    SERVICE_LIST: '서비스 리스트',
    CATEGORY_LIST: '카테고리 리스트',

    SUBS_LIST: '구독 리스트',
    SUBS_ADD_BUTTON: '+ 구독 추가',
    SAVE: '저장',
    SUBMIT: '등록',
    CANCEL: '취소',
    // Service Screen
    SERVICE_ADD_BUTTON: '+ 서비스 추가',
    // Category Screen
    CATEGORY_ADD_BUTTON: '+ 카테고리 추가',
    // Service Modal
    ICON: '아이콘',
    THUMBNAIL: '썸네일',
    IMAGE: '이미지',
    SERVICE_NAME_KEY: '키값',
    WEB_PAGE: 'Web url',
    ANDROID_PLAY_STORE_URL: '안드로이드 url',
    IOS_APP_STORE_URL: 'IOS url',
    MEMO: '부가 설명',
    PNG_JPG_GIF_ONLY: 'png, jpg파일과 gif파일만 업로드 가능합니다',
    INPUT_REQUIRED: '의 입력이 필요합니다',
    INPUT_MAX_VALIDATION: '해당 길이 이상으로 입력될 수 없습니다:',
    INPUT_TEXT_TO_SEARCH: '검색어를 입력하세요.',
    // Service List
    SERVICE_LIST_TABLE_LABEL_ID: 'ID',
    SERVICE_LIST_TABLE_LABEL_NAME: '영문',
    SERVICE_LIST_TABLE_LABEL_NAME_KR: '국문',
    SERVICE_LIST_TABLE_LABEL_THUMBNAIL: '썸네일',
    SERVICE_LIST_TABLE_LABEL_SETTING: '설정',
    PREVIEW_ICON: '미리보기',
    // Category List
    CATEGORY_LIST_TABLE_LABEL_ID: 'ID',
    CATEGORY_LIST_TABLE_LABEL_KEY: '키',
    CATEGORY_LIST_TABLE_LABEL_DESCRIPTION: '설명',
    CATEGORY_LIST_TABLE_LABEL_RELEASED: 'Released',
    CATEGORY_LIST_TABLE_LABEL_SETTING: '설정',
    // Service Group
    SERVICE_GROUP_ADD_BUTTON: '+ 서비스 그룹 추가',
    SERVICE_GROUP_LIST: '서비스 그룹 리스트',
    SERVICE_GROUP_LIST_TABLE_LABEL_ID: 'ID',
    SERVICE_GROUP_LIST_TABLE_LABEL_KEY: '키',
    SERVICE_GROUP_LIST_TABLE_LABEL_DESCRIPTION: '설명',
    SERVICE_GROUP_LIST_TABLE_LABEL_SETTING: '설정',
    // Service Details
    SERVICE_LABEL_ID: 'ID',
    SERVICE_LABEL_NAME: 'Subs 키',
    SERVICE_LABEL_PRICE_PLAN_CURRENCY: '통화',
    SERVICE_LABEL_PRICE: '가격',
    SERVICE_LABEL_PERIOD_UNIT: '갱신단위',
    SERVICE_LABEL_PERIOD_TYPE: '갱신주기',
    SERVICE_LABEL_SETTING: '설정',
    // Subs modal
    SUBS_MODAL_TITLE: '구독 추가',
    SUBS_MODAL_NAME: '구독명',
    SUBS_MODAL_CURRENCY: '통화',
    SUBS_MODAL_PRICE: '가격',
    SUBS_MODAL_PAYMENT_DATE_UNIT: '갱신',
    SUBS_MODAL_CASE: '케이스',
    SUBS_MODAL_MEMO: '메모',
    // Category modal
    CATEGORY_MODAL_CREATE_TITLE: '카테고리 추가',
    CATEGORY_MODAL_UPDATE_TITLE: '카테고리 수정',
    CAT_MODAL_KEY: '키',
    CAT_MODAL_DESC: '설명',
    CAT_MODAL_RELEASED: 'Released',
    // Service Group modal
    SERVICE_GROUP_MODAL_CREATE_TITLE: '카테고리 추가',
    SERVICE_GROUP_MODAL_UPDATE_TITLE: '카테고리 수정',
    SERVICE_GROUP_MODAL_KEY: '키',
    SERVICE_GROUP_MODAL_DESC: '설명',
  },
  // ja: { // Japanese
  // },
  // zh: { // Chinese
  // },
  // es: { // Spanish
  // },
  // fr: { // French
  // },
  // id: { // Indonesian
  // },
};

const DEFAULT_LANG = 'en';
let lang = DEFAULT_LANG;

if (navigator) {
  // lang = navigator.language.substr(0, 2);
  lang = 'ko'; // set it for test
}

export const getString = (inputStr: string) => {
  let str = STRINGS[lang]
    ? STRINGS[lang][inputStr]
    : STRINGS[DEFAULT_LANG][inputStr];

  if (!str) {
    // console.warn(`${inputStr} doesn't exist on STRING`);
    str = inputStr;
  }
  return str;
};

export default STRINGS;
