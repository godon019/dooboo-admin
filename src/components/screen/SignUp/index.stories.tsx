// import SignUp, { PureSignUp } from '.';

// import { FrameDecorator } from '../../../../.storybook/decorators';
// import React from 'react';
// import { storiesOf } from '@storybook/react';

// // const Good = WithSignUpApollo(SignUp);
// storiesOf('SignUp', module)
//   .addDecorator(FrameDecorator)
//   .add('default', () => (
//     <PureSignUp
//       onSubmitSuccess={({ email, password }) => {
//         console.log(`onSubmit, email: ${email}, password: ${password}`);
//       }}
//     />
//   ))
//   .add('WithApollo', () => <SignUp />);
