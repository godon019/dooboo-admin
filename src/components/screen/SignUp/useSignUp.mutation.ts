import { AdminSignUp, AuthPayload } from '../../../types';

import { GQL } from '../../../utils/Constants';
import gql from 'graphql-tag';
import { propertyNamesOf } from '../../../utils/withApollo';
import { useMutation } from 'react-apollo';
import { useState } from 'react';

interface Variable {
  admin: AdminSignUp;
}
interface Data {
  signUpAdmin: AuthPayload;
}
const variableProperty = propertyNamesOf<Variable>()('admin');
const dataProperty = propertyNamesOf<Data>()('signUpAdmin');

export const GRAPH_QL = gql`
  mutation signUpAdminMutation($admin: AdminCreateInput!) {
    signUpAdmin(admin: $admin) {
      token
      admin {
        id
        name
        email
      }
    }
  }
`;

const useCustomMutation = () => {
  const [_mutate, { error, loading, called }] = useMutation<Data, Variable>(
    GRAPH_QL,
  );
  const [result, setResult] = useState(null);

  const mutate = async ({ email, password, name }: AdminSignUp) => {
    const { data } = await _mutate({
      refetchQueries: () => [GQL.categories],
      variables: {
        [variableProperty]: {
          // be careful not to include any unnecessary property, coz typing is not enough to filter them out
          email,
          password,
          name,
        },
      },
    });

    if (data && data[dataProperty]) {
      setResult(data[dataProperty]);
    }
  };

  return { mutate, result, error, loading, called };
};

export { useCustomMutation as useSignUp };
