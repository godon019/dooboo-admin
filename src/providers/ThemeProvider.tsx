import React, { useContext } from 'react';
import { ThemeType, createTheme } from '../theme';

import { AppContext } from '../contexts';
import { ThemeProvider as StyledComponentThemeProvider } from 'styled-components';

interface Props {
  children?: any;
}
const ThemeProvider = (props: Props) => {
  const fromAppProvider = useContext(AppContext);
  /**
   * StyledComponentThemeProvider is normally used with AppProvider as its parent
   * and get `theme` value from AppProvider
   * if AppProvider doesn't exist
   * then use ThemeType.LIGHT as default `theme`
   */
  const theme = fromAppProvider ? fromAppProvider.state.theme : ThemeType.LIGHT;
  return (
    <StyledComponentThemeProvider theme={createTheme(theme)}>
      {props.children}
    </StyledComponentThemeProvider>
  );
};

export default ThemeProvider;
