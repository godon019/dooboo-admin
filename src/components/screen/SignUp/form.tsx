import { AdminSignUp, AdminSignUpForm } from '../../../types';
import React, { Component, useState } from 'react';

import Button from '../../shared/Button';
import { FormikProps } from 'formik';
import Input from '../../shared/Input';
import { getString } from '../../../../STRINGS';
import styled from 'styled-components';

const Form = styled.form`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

const Row = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-bottom: 1.5em;
`;
const Label = styled.label`
  font-size: 1rem;
  font-weight: bold;
`;

const ErrorMessage = styled.div`
  color: red;
  font-size: 0.7em;
`;

const SignInForm: React.FC<
  React.PropsWithChildren<FormikProps<AdminSignUpForm>>
> = ({
  values,
  errors,
  touched,
  handleChange,
  handleBlur,
  handleSubmit,
  isSubmitting,
}) => (
  <Form onSubmit={handleSubmit}>
    <Row>
      <Label htmlFor='email'>{getString('EMAIL') + ':'}</Label>
      <Input
        type='email'
        name='email'
        data-testid='email'
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.email}
      />
      {errors.email && touched.email && (
        <ErrorMessage data-testid='error-email'>{errors.email}</ErrorMessage>
      )}
    </Row>
    <Row>
      <Label htmlFor='name'>{'이름:'}</Label>
      <Input
        type='name'
        name='name'
        data-testid='name'
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.name}
      />
      {errors.name && touched.name && (
        <ErrorMessage data-testid='error-name'>{errors.name}</ErrorMessage>
      )}
    </Row>
    <Row>
      <Label>{getString('PASSWORD') + ':'}</Label>
      <Input
        type='password'
        name='password'
        data-testid='password'
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.password}
      />
      {errors.password && touched.password && (
        <ErrorMessage data-testid='error-password'>
          {errors.password}
        </ErrorMessage>
      )}
    </Row>
    <Row>
      <Label>{getString('PASSWORD_CONFIRM') + ':'}</Label>
      <Input
        type='password'
        name='passwordConfirm'
        data-testid='passwordConfirm'
        onChange={handleChange}
        onBlur={handleBlur}
        value={values.passwordConfirm}
      />
      {errors.passwordConfirm && touched.passwordConfirm && (
        <ErrorMessage data-testid='error-passwordConfirm'>
          {errors.passwordConfirm}
        </ErrorMessage>
      )}
    </Row>
    <Button
      type='submit'
      text={getString('SIGNUP')}
      data-testid='submit'
      disabled={isSubmitting}
    />
  </Form>
);
export default SignInForm;
