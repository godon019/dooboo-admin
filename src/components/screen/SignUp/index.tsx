import { AdminSignUp, AdminSignUpForm } from '../../../types';
import React, { Component, useState } from 'react';

import { Formik } from 'formik';
import SignInForm from './form';
import { getString } from '../../../../STRINGS';
import styled from 'styled-components';

const ContentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 0 35%;
`;
const TitleText = styled.div`
  font-size: 1.3rem;
  font-weight: bold;
  width: 100%;
  text-align: left;
  margin-bottom: 1.5rem;
`;

export interface SignUpProps {
  onSubmitSuccess: (submitValues: AdminSignUp) => void; // test purpose
}

const SignUp: React.FC<SignUpProps> = (props) => {
  return (
    <ContentWrapper>
      <TitleText>{getString('SIGNUP')}</TitleText>
      <Formik<AdminSignUpForm>
        initialValues={{
          email: '',
          name: '',
          password: '',
          passwordConfirm: '',
        }}
        validate={(values) => {
          const errors: {
            email?: string;
            name?: string;
            password?: string;
            passwordConfirm?: string;
          } = {};

          // email validation
          if (!values.email) {
            errors.email = getString('EMAIL_HINT');
          } else if (
            !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
          ) {
            errors.email = getString('EMAIL_ERROR');
          }

          if (!values.name) {
            errors.name = '이름을 입력해 주세요';
          }

          // password validation
          if (!values.password) {
            errors.password = getString('PASSWORD_HINT');
          }
          // password_confirm validation
          if (!values.passwordConfirm) {
            errors.passwordConfirm = getString('PASSWORD_HINT');
          }
          if (
            values.password &&
            values.passwordConfirm &&
            values.password !== values.passwordConfirm
          ) {
            errors.password = getString('PASSWORD_NOT_MATCH');
            errors.passwordConfirm = getString('PASSWORD_NOT_MATCH');
          }
          return errors;
        }}
        onSubmit={(values, { setSubmitting }) => {
          if (props.onSubmitSuccess) {
            props.onSubmitSuccess(values);
          }
          setSubmitting(false);
        }}
        component={(props) => <SignInForm {...props} />}
      />
    </ContentWrapper>
  );
};
export default SignUp;
