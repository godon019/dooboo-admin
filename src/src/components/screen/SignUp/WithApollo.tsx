import { Admin, AdminSignUp, AuthPayload } from '../../../types';
import React, { useContext, useEffect } from 'react';
import { getDisplayName, useLoadingOverlay } from '../../../utils/withApollo';

import { DisplayState } from '../../../utils/DisplayState';
import { ROUTE } from '../../../utils/Constants';
import { RouteContext } from '../../../providers/RouteProvider';
import { SignUpProps } from '.';
import gql from 'graphql-tag';
import { useMutation } from 'react-apollo';
import { useSignUp } from './useSignUp.mutation';

type PropsToExclude = Pick<SignUpProps, 'onSubmitSuccess'>;

export const SIGN_UP_ADMIN = gql`
  mutation signUpAdminMutation($admin: AdminCreateInput!) {
    signUpAdmin(admin: $admin) {
      token
      admin {
        id
        email
      }
    }
  }
`;

export function WithSignUpApollo<T extends PropsToExclude>(
  WrappedComponent: React.FC<T>,
) {
  const hocComponent = (props: Omit<T, keyof PropsToExclude>) => {
    const { mutate, ...rest } = useSignUp();
    const { route } = useContext(RouteContext);
    const { result, error, loading } = rest;
    useLoadingOverlay(`SignUp Mutation`, loading, error);

    useEffect(() => {
      if (result) {
        alert(
          '회원가입이 완료되었습니다. 관리자의 승인 이후에 로그인이 가능합니다',
        );
        route.history.push(ROUTE.signIn);
      }
    }, [result]);

    useEffect(() => {
      if (error) {
        alert(error.message);
      }
    }, [error]);

    const propsToExclude: PropsToExclude = {
      onSubmitSuccess: (values) => {
        mutate(values);
      },
    };
    const wholeProps: T = {
      ...props,
      ...propsToExclude,
    } as T;

    return (
      <div>
        <WrappedComponent {...wholeProps} />
        <DisplayState {...rest} />
      </div>
    );
  };

  hocComponent.displayName = `WithSignUpApollo(${getDisplayName(
    WrappedComponent,
  )})`;

  return hocComponent;
}
