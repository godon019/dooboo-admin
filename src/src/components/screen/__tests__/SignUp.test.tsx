import * as renderer from 'react-test-renderer';

import {
  fireEvent,
  getByTestId,
  queryByTestId,
  render,
  wait,
} from '@testing-library/react';

import React from 'react';
import SignUp from '../SignUp';
import ThemeProvider from '../../../providers/ThemeProvider';
import { getString } from '../../../../STRINGS';

const props = {
  onSubmitSuccess: jest.fn(),
};

describe('[SignUp] render', () => {
  it('renders without crashing', () => {
    const rendered = renderer
      .create(
        <ThemeProvider>
          <SignUp />
        </ThemeProvider>,
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});

describe('[SignUp] Interaction', () => {
  const component = (
    <ThemeProvider>
      <SignUp {...props} />
    </ThemeProvider>
  );
  let renderResult: any;

  beforeAll(() => {
    renderResult = render(component);
  });

  it('[onSubmit] of Formik should not be called when submit button is clicked with empty inputs', async () => {
    const emailInput = renderResult.getByTestId('email');
    expect(emailInput.value).toBe('');
    const passwordInput = renderResult.getByTestId('password');
    expect(passwordInput.value).toBe('');
    const passwordConfirmInput = renderResult.getByTestId('passwordConfirm');
    expect(passwordConfirmInput.value).toBe('');
    const submitButton = renderResult.getByTestId('submit');
    fireEvent.click(submitButton);

    // formik is called asynchronously
    await wait(() => {
      expect(props.onSubmitSuccess).toHaveBeenCalledTimes(0);
    });
  });

  it('[onSubmit] of Formik should be called when [submit] button is clicked with valid input values', async () => {
    const email = 'whatssub123@gmail.com';
    const password = '1234';
    const passwordConfirm = '1234';

    const emailInput = renderResult.getByTestId('email');
    fireEvent.change(emailInput, { target: { value: email } });
    expect(emailInput.value).toBe(email);

    const passwordInput = renderResult.getByTestId('password');
    fireEvent.change(passwordInput, { target: { value: password } });
    expect(passwordInput.value).toBe(password);

    const passwordConfirmInput = renderResult.getByTestId('passwordConfirm');
    fireEvent.change(passwordConfirmInput, {
      target: { value: passwordConfirm },
    });
    expect(passwordConfirmInput.value).toBe(password);

    const submitButton = renderResult.getByTestId('submit');
    fireEvent.click(submitButton);

    // formik is called asynchronously
    await wait(() => {
      expect(props.onSubmitSuccess).toHaveBeenCalledTimes(1);
    });
  });

  it('has email [error] massage when email is not valid', async () => {
    const email = 'notVailidEmail';

    // error message dosen't exist at first
    expect(renderResult.queryByTestId('error-email')).toBeNull();

    const emailInput = renderResult.getByTestId('email');
    fireEvent.focus(emailInput);
    fireEvent.change(emailInput, { target: { value: email } });
    fireEvent.blur(emailInput);

    await wait(() => {
      const errorMessage = renderResult.getByTestId('error-email');
      expect(errorMessage).toBeDefined();
      expect(errorMessage.textContent).toBe(getString('EMAIL_ERROR'));
    });
  });

  it('has password and passwordConfirm [error] massage when password and passwordConfirm is not equal', async () => {
    const password = '1234';
    const passwordConfirm = '12345';

    // re-render component
    const errorPassword = renderResult.queryByTestId('error-password');
    const errorPasswordConfirm = renderResult.queryByTestId(
      'error-passwordConfirm',
    );

    // error message dosen't exist at first
    expect(errorPassword).toBeNull();
    expect(errorPasswordConfirm).toBeNull();

    const passwordInput = renderResult.getByTestId('password');
    fireEvent.focus(passwordInput);
    fireEvent.change(passwordInput, { target: { value: password } });
    fireEvent.blur(passwordInput);
    const passwordConfirmInput = renderResult.getByTestId('passwordConfirm');
    fireEvent.focus(passwordConfirmInput);
    fireEvent.change(passwordConfirmInput, {
      target: { value: passwordConfirm },
    });
    fireEvent.blur(passwordConfirmInput);

    await wait(() => {
      const errorPassword = renderResult.queryByTestId('error-password');
      const errorPasswordConfirm = renderResult.queryByTestId(
        'error-passwordConfirm',
      );
      expect(errorPassword).toBeDefined();
      expect(errorPasswordConfirm).toBeDefined();
      expect(errorPassword.textContent).toBe(getString('PASSWORD_NOT_MATCH'));
      expect(errorPasswordConfirm.textContent).toBe(
        getString('PASSWORD_NOT_MATCH'),
      );
    });
  });
});
