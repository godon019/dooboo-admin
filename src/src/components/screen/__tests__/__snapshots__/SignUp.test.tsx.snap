// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SignUp] render renders without crashing 1`] = `
<div
  className="sc-chPdSV bvcEDN"
>
  <div
    className="sc-jTzLTM ihSsTl"
  >
    <div
      className="sc-fjdhpX jeAUlG"
    >
      <img
        className="sc-cSHVUG gguuzs"
        src="https://github.com/whatssub/whatssub-mobile/blob/master/assets/splash.gif?raw=true"
      />
      <span
        className="sc-jzJRlG kgEAHB"
      >
        | Admin Page
      </span>
    </div>
    <div
      className="sc-kAzzGY iQvEvC"
    />
  </div>
  <div
    className="sc-kgoBCf bcGaPZ"
  >
    <div
      className="sc-kGXeez iNpFKN"
    >
      회원가입
    </div>
    <form
      className="sc-dnqmqq gZXnwY"
      onSubmit={[Function]}
    >
      <div
        className="sc-iwsKbI fJNkWy"
      >
        <label
          className="sc-gZMcBi gfQqjX"
          htmlFor="email"
        >
          이메일:
        </label>
        <div
          className="sc-bZQynM bYznNk"
        >
          <input
            className="sc-gzVnrw jzxJOR"
            data-testid="email"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        className="sc-iwsKbI fJNkWy"
      >
        <label
          className="sc-gZMcBi gfQqjX"
        >
          비밀번호:
        </label>
        <div
          className="sc-bZQynM bYznNk"
        >
          <input
            className="sc-gzVnrw jzxJOR"
            data-testid="password"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        className="sc-iwsKbI fJNkWy"
      >
        <label
          className="sc-gZMcBi gfQqjX"
        >
          비밀번호 확인:
        </label>
        <div
          className="sc-bZQynM bYznNk"
        >
          <input
            className="sc-gzVnrw jzxJOR"
            data-testid="passwordConfirm"
            name="passwordConfirm"
            onBlur={[Function]}
            onChange={[Function]}
            type="password"
            value=""
          />
        </div>
      </div>
      <button
        className="sc-bdVaJa bEdxjs"
        data-testid="submit"
        disabled={false}
        style={
          Object {
            "height": "60px",
          }
        }
        type="submit"
      >
        <div>
          <span
            className="sc-bxivhb koiFyU"
            style={Object {}}
          >
            회원가입
          </span>
        </div>
      </button>
    </form>
  </div>
</div>
`;
